spark3/src/main/java/org/apache/iceberg/spark/source/SparkBatchScan.java

```diff
@@ -112,6 +124,8 @@ public Batch toBatch() {
   @Override
   public StructType readSchema() {
     if (readSchema == null) {
+      Preconditions.checkArgument(readTimestampWithoutZone || !hasTimestampWithoutZone(expectedSchema),
+              "Spark does not support timestamp without time zone fields");
```

Comment:
Same comment as above on the error message, since it's gonna be in a few places we should probably have the error message as a constant too.

Reply:
addressed it